java/client/src/main/java/glide/api/models/commands/WeightAggregateOptions.java

```diff
@@ -0,0 +1,64 @@
+/** Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0 */
+package glide.api.models.commands;
+
+import glide.api.commands.SortedSetBaseCommands;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.Builder;
+import lombok.Singular;
+
+/**
+ * Optional arguments to {@link SortedSetBaseCommands#zunionstore}
+ *
+ * @see <a href="https://redis.io/commands/zunionstore/">redis.io</a> for more details.
+ */
+@Builder
+public final class WeightAggregateOptions {
```

Comment:
Consider two things:
1. Split this into 2 independent options; you need to add more function signatures then
2. Implement static factory constructors which will build options with weights and/or with aggregation
3. Both

Reply:
What do you not like about this I am confused? Are you worried that some functions will use weights and not aggregation or vice a versa I just followed the other options we had.